### PR TITLE
Optimise github actions

### PR DESCRIPTION
**Pull request by rkachowski**
Optimises the workflow by aggressively caching dependencies. 

Notably :

* Cache elixir dependencies
* Cache npm node_modules
* Cache dialyzer persistent lookup table (plt)

These caches are keyed on their respective lock files, for npm this is the `package-lock.json` files in `explorer` and `block_scout_web` apps, for dialyzer and elixir dependencies this is `mix.lock`. These caches will be rebuilt upon changes to npm modules or elixir dependencies, which will result in an initially longer build time but will be persisted across test runs.

Initial results seem to show a reduction in test run time from ~23 minutes to around 14 minutes. Currently the test runs are bounded by the `explorer` test suite, specifically smart contract verification functionality. Further reductions would likely involve significant code level changes which will come in further pull requests.

### CODE REVIEW

**Review by lvpeschke — APPROVED** [review]

File: .github/workflows/config.yml [hunk 1]
```diff
@@ -11,183 +11,310 @@ on:
...
           otp-version: '23.3.4.1'
           elixir-version: '1.11.4'
-      - run: mix local.hex --force
-      - run: mix local.rebar --force
-      - run: mix deps.get
-      - run: mix compile
-      - run: make
-        working-directory: "deps/libsecp256k1"
+
+      - name: Restore Mix Deps Cache
```
Comment (carterqw2) [inline .github/workflows/config.yml:118]: Did you look into composite actions that could allow us extract these restore cache steps into a local action and avoid repetition like described [here](https://github.com/actions/runner/issues/646#issuecomment-901336347)?

Comment (rkachowski) [reply .github/workflows/config.yml:118]: That seems pretty cool

Comment (rkachowski) [reply .github/workflows/config.yml:118]: It looks like we can't use it right now due to this feature not supporting conditionals. We actually want to do the same as  https://github.com/actions/runner/issues/834#issuecomment-901847269 and create a wrapper around the cache action

Comment (carterqw2) [reply .github/workflows/config.yml:118]: Ah, true, missed that.

**Pull request merged**